src/ui/QueryEditor.tsx

```diff
@@ -0,0 +1,152 @@
+import React from 'react';
+import { QueryEditorProps, SelectableValue } from '@grafana/data';
+import { DataSource } from '../DataSource';
+import { CmkQuery, DataSourceOptions, ResponseDataAutocomplete } from '../types';
+import { cloneDeep } from 'lodash';
+import { InlineFieldRow, VerticalGroup } from '@grafana/ui';
+import { FilterEditor, Select, StringSelect } from './components';
+import { createAutocompleteConfig, Presentation } from './autocomplete';
+import { defaultRequestSpec, RequestSpec } from '../RequestSpec';
+import { titleCase } from '../utils';
+
+type GraphKind = 'template' | 'metric';
+
+type Props = QueryEditorProps<DataSource, CmkQuery, DataSourceOptions>;
+
+export const QueryEditor = (props: Props): JSX.Element => {
+  const { onChange, onRunQuery, datasource, query } = props;
+  const requestSpec = query.requestSpec || defaultRequestSpec;
+
+  const editionMode = datasource.getEdition() ?? 'RAW';
+
+  function isMinimalRequest(spec: RequestSpec) {
+    return [spec.site, spec.host_name, spec.graph_type, spec.graph].map(Boolean).reduce((x, y) => x && y);
+  }
+
+  const autocomplete =
+    (ident: string) =>
+    async (value = '') => {
+      const response = await props.datasource.autocompleterRequest<ResponseDataAutocomplete>(
+        'ajax_vs_autocomplete.py',
+        createAutocompleteConfig(requestSpec, ident, value)
+      );
+      return response.data.result.choices.map(([value, label]: [string, string]) => ({
+        value,
+        label,
+      }));
+    };
+
+  const labelAutocomplete = async (value: string) => {
+    const response = await props.datasource.autocompleterRequest<Array<{ value: string }>>(
+      'ajax_autocomplete_labels.py',
+      {
+        world: 'core',
+        search_label: value,
+      }
+    );
+    return response.data.result.map((val: { value: string }) => ({ value: val.value, label: val.value }));
+  };
+
+  const completeTagChoices = async (tagGroupId: string, value: string) => {
+    const response = await props.datasource.autocompleterRequest<ResponseDataAutocomplete>('ajax_vs_autocomplete.py', {
+      ident: 'tag_groups_opt',
+      params: { group_id: tagGroupId, strict: true },
+      value: value,
+    });
+    return response.data.result.choices.map(([value, label]: [string, string]) => ({
+      value,
+      label,
+    }));
+  };
+
+  const presentationCompleter = async () =>
+    [
+      { value: 'lines', label: 'Lines' },
+      { value: 'sum', label: 'Sum' },
+      { value: 'average', label: 'Average' },
+      { value: 'min', label: 'Minimum' },
+      { value: 'max', label: 'Maximum' },
+    ] as Array<SelectableValue<Presentation>>;
```

Comment:
hm. seems like `Presentation` is not checked in this setting, also not if you put it into a variable that is type like that. not sure how to fix it...